snapshot/snapshotter.go

```diff
@@ -135,3 +142,41 @@ func (cs *containerdSnapshotter) View(ctx context.Context, key, parent string, o
 	}
 	return cs.returnMounts(mf)
 }
+
+var redirectDirOption string
+var redirectDirOptionOnce sync.Once
+
+func getRedirectDirOption() string {
+	redirectDirOptionOnce.Do(func() {
+		if _, err := os.Stat("/sys/module/overlay/parameters/redirect_dir"); err != nil {
+			redirectDirOption = "" // redirect_dir unsupported
+			return
+		}
+		if userns.RunningInUserNS() {
+			redirectDirOption = "nofollow" // follow is not allowed in user ns
+			return
+		}
+		redirectDirOption = "off" // disable redirect_dir to avoid broken diff
+	})
+	return redirectDirOption
+}
+
+func mountsRedirectDirOption(mounts []mount.Mount, redirectDirOption string) (ret []mount.Mount) {
```

Comment:
`addRedirectDirOption` might be better name.

Or maybe even `addMountOption(mounts, k, v)`